### DISCUSSION TITLE
How to limit packages affected by `ignoreUnstable`?

### DISCUSSION THREAD

**Discussion by qubitz** [renovatebot/renovate#20388] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

## Questions
Does `ignoreUnstable` only work at the root level? Even if it is at the root level, does it apply to all packages? We want to scope the `ignoreUnstable` option only for our polyrepo packages using `packageRules`. However, though it is at the root level in our current configurations, it seems only packages hosted in registries defined in `hostRules` are affected. Is this accurate?

## Context
Our team has a polyrepo system where we propagate preview packages from low level repositories to their higher level dependents. We are currently using `ignoreUnstable` to implement "stable" -> "unstable" bump behavior to quickly develop on the next dependent downstream.

### What you tried so far.

```json
{
    "packageRules": [
        {
            "matchPackagePrefixes": [
                "Link"
            ],
            "ignoreUnstable": false // hoped to scope only to these packages
        }
    ],
    "nuget": {
        "registryUrls": [
            "https://nuget.pkg.github.com/igt-all/index.json",
            "https://api.nuget.org/v3/index.json"
        ]
    },
    "hostRules": [
        {
            "matchHost": "https://nuget.pkg.github.com/igt-all/",
            "hostType": "nuget",
            "username": "username",
            "encrypted": {
                "password": "--snip--"
            }
        }
     ]
}
```

### Relevant debug logs

Lost to time about a week ago. Happy to work towards a reproducible configuration with logs if indeed `ignoreUnstable` is not behaving as expected.

**Comment by rarkins** [comment, marked as answer]
`ignoreUnstable` should work per-package as you wanted. Set up a reproduction repo if you want us to look into it

**Reply by JavierSegoviaCordoba** [reply]
@qubitz did you create an issue for this one?

**Reply by JavierSegoviaCordoba** [reply]
@rarkins is there any workaround with package rules right now? I want to force that a package always update to latest unstable version even if the current one is stable and there is another higher stable version

**Reply by rarkins** [reply]
You could perhaps define an `allowedVersions` pattern which only satisfies unstable versions
